tests/test_indexes.py

```diff
@@ -312,6 +316,20 @@ def test_range_index_union():
         ),
         pd.Index,
     )
+    check(
+        assert_type(
+            pd.RangeIndex(0, 10).union(["a", "b", "c"], sort=True),
+            Union[pd.Index, "pd.Index[int]", pd.RangeIndex],
+        ),
+        pd.Index,
+    )
+    check(
+        assert_type(
+            pd.RangeIndex(0, 10).union(["a", "b", "c"], sort=False),
+            Union[pd.Index, "pd.Index[int]", pd.RangeIndex],
+        ),
```

Comment:
This doesn't seem right, and I think you'll need some specific overloads here for `RangeIndex` (and `Sequence[int]` for that matter).   If `RangeIndex` has a set operation with either a `Sequence[int]`, `Series[int]`, `Index[int]` or `RangeIndex`, the result is `Union[RangeIndex, Index[int]]`.  With any other type, it is just `Index`

Reply:
Indeed that was the wrong copy paste, this is fixed.